### ISSUE TITLE
CI tests: MaxListenersExceededWarning: Possible EventEmitter memory leak detected.

### ISSUE THREAD

**Issue by juergba** [mochajs/mocha#4106]
In the log files of our CI tests, following warnings can be found:

> MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 uncaughtException listeners added to [process].

Please investigate, explain and evtl. fix those warnings.

**Comment by adam-edison** [comment]
Confirmed I've been seeing this bug for the last 3 weeks. Running on node 10, using mocha version 6.2.2. This outputs to the console while tests are running using either `mocha` or `mocha --watch`.

```
(node:13167) MaxListenersExceededWarning: 
Possible EventEmitter memory leak detected. 
11 uncaughtException listeners added to [process]. 
Use emitter.setMaxListeners() to increase limit
```

Suggested cause:
- listeners are not being removed after they have been used and no longer needed

Many other libraries have faced this issue. See winston:
* https://github.com/node-ts/logger/pull/13

**Comment by jehy** [comment]
I described this bug in details here: https://github.com/mochajs/mocha/issues/4171

**Comment by juergba** [comment]
fixed by #4147.

Our CI tests still log a few warnings, but they are test specific and will be improved later.
